enterprise/server/hostedrunner/hostedrunner.go

```diff
@@ -217,10 +198,13 @@ func (r *runnerService) createAction(ctx context.Context, req *rnpb.RunRequest,
 }
 
 func (r *runnerService) withCredentials(ctx context.Context, req *rnpb.RunRequest) (context.Context, error) {
-	apiKey, err := r.lookupAPIKey(ctx)
-	if err != nil {
-		return nil, err
+	apiKeys := metadata.ValueFromIncomingContext(ctx, authutil.APIKeyHeader)
```

Comment:
What do you think about using GetAPIKeyForInternalUseOnly to have parity with the workflow code? 
I don't love the idea of looking at the API key header directly because if we make changes to the auth layer, this could potentially break since we generally don't expect non-auth code to be poking at these headers.

Reply:
Ack - changed